### DISCUSSION TITLE
How **not** to publish to npm?

### DISCUSSION THREAD

**Discussion by hhimanshu** [semantic-release/semantic-release#1895] [Q&A]
Hello

Firstly, **THANK YOU** for `semantic-release`, it is such a nice solution that helps do the release. It has takes so many of my questions out of question ;-)

I am working on an app that is deployed on Vercel. The code is [available on GitHub](https://github.com/hhimanshu/typescript-serverless-api-vercel).
I used semantic-release for release, which also did publish to npm([package on npmjs](https://www.npmjs.com/package/@hhimanshu/typescript-serverless-vercel))

Now since this is an app and is supposed to be deployed, I am thinking it may not be a good idea to publish it as a package on npm (Am I thinking right?)
I only need to make a release using semantic-releases and that is happening ([see releases](https://github.com/hhimanshu/typescript-serverless-api-vercel/releases)).

What I want is **not** to publish on npm. How can I achieve that?

Kindly guide me

**Comment by gr2m** [comment]
in your package.json, add

```json
  "private": true
```

I'm doing this all the time for my apps, GitHub Actions, etc :)

See https://github.com/semantic-release/npm/#options for details

**Comment by hhimanshu** [comment]
I also learned from documentation to override the configuration for the `@semantic-release/npm` plugin as below in `package.json`
```json
    "plugins": [
      [
        "@semantic-release/npm",
        {
          "npmPublish": false
        }
      ]
    ]
  }
```
Anf this worked as well. [Here is the GitHub Action log file](https://github.com/hhimanshu/typescript-serverless-api-vercel/runs/2370264088?check_suite_focus=true)
